src/handlers/handle-transfers.ts

```diff
@@ -0,0 +1,105 @@
+import { log, Address, BigInt, Bytes } from "@graphprotocol/graph-ts";
+import { Transfer as TransferEvent } from "../../generated/SingleEditionMintableCreator/SingleEditionMintable"
+import { FreeNFTBalance, FreeNFTDrop } from "../../generated/schema";
+
+
+let NULL_ADDRESS = "0x0000000000000000000000000000000000000000";
+
+
+function updateBalance(
+  event: TransferEvent, 
+  drop: FreeNFTDrop, 
+  address: Address, 
+  isSender: boolean = false,
+): void {
+  if (address.toHexString() == NULL_ADDRESS) {
+    log.info(
+      "Ignoring null address for balance update in tx {} log index {}, isSender = {}",
+      [
+        event.transaction.hash.toHexString(),
+        event.transactionLogIndex.toString(),
+        isSender.toString(),
+      ],
+    );
+    
+    return;
+  }
+
+  let id = drop.id.toHexString() + "-" + address.toHexString(); 
+  
+  let balance = FreeNFTBalance.load(id);
+
+  if (balance == null) {
+    balance = new FreeNFTBalance(id);
+
+    balance.address = address;
+    balance.quantity = BigInt.fromI32(0);
+    balance.drop = drop.id;
+  }
+  
+  if (balance.address != address) {
+    log.error(
+      "Balance with id {} has owner address {}, which does not match address {}",
+      [
+        balance.id,
+        balance.address.toHexString(),
+        address.toHexString(),
+      ]
+    );
+  }
+  
+  if (isSender) {
+    balance.quantity = balance.quantity.minus(BigInt.fromI32(1));
+  }
+  
+  else {
+    balance.quantity = balance.quantity.plus(BigInt.fromI32(1));
+  }
+
+  balance.updatedAt = event.block.timestamp;
+  balance.save();
+}
+
+
+export default function handleTransfer(event: TransferEvent): void {
+  if (!event.address) {
+    log.critical(
+      "No contract address in tx {}",
+      [event.transaction.hash.toHex()]
+    );
+
+    return;
+  }
+
+  let editionAddress = event.address;
+  
+  let drop = FreeNFTDrop.load(Bytes.fromHexString(editionAddress.toHexString()));
+  
+  if (drop == null) {
+    log.critical(
```

Comment:
wouldn't that happen for every transfer from a non-Showtime drop? `log.critical` terminates the subgraph, I think we should either just ignore it or `log.debug` and move on

Reply:
no! the handler is only triggered when a drop is created and if it matches the parent drop. not sure how that works but it's some template magic from the graph!

Reply:
for posterity here's the docs on templates https://thegraph.com/docs/en/developing/creating-a-subgraph/#data-source-templates-for-dynamically-created-contracts

Reply:
very cool!

docs: https://thegraph.com/docs/en/developing/creating-a-subgraph/#data-source-templates-for-dynamically-created-contracts